Cython/Compiler/ModuleNode.py

```diff
@@ -1596,22 +1596,25 @@ def generate_new_function(self, scope, code, cclass_entry):
             code.putln("PyObject *o = %s(t, a, k);" % tp_new)
         else:
             code.putln("PyObject *o;")
-            code.putln("#if CYTHON_COMPILING_IN_LIMITED_API")
-            code.putln("allocfunc alloc_func = (allocfunc)PyType_GetSlot(t, Py_tp_alloc);")
-            code.putln("o = alloc_func(t, 0);")
-            code.putln("#else")
             if freelist_size:
                 code.globalstate.use_utility_code(
                     UtilityCode.load_cached("IncludeStringH", "StringTools.c"))
                 if is_final_type:
-                    type_safety_check = ''
+                    heap_type_safety_check = type_safety_check = ''
                 else:
                     type_safety_check = ' & (int)(!__Pyx_PyType_HasFeature(t, (Py_TPFLAGS_IS_ABSTRACT | Py_TPFLAGS_HEAPTYPE)))'
+                    heap_type_safety_check = ' & (int)(!__Pyx_PyType_HasFeature(t, Py_TPFLAGS_IS_ABSTRACT))'
                 obj_struct = type.declaration_code("", deref=True)
                 code.putln("#if CYTHON_USE_FREELISTS")
-                code.putln(
-                    "if (likely((int)(%s > 0) & (int)(t->tp_basicsize == sizeof(%s))%s)) {" % (
-                        code.name_in_slot_module_state(freecount_name), obj_struct, type_safety_check))
+                code.putln(f"if (likely((int)({code.name_in_slot_module_state(freecount_name)} > 0)) &")
```

Comment:
`likely` has been rebracketed here just to cover the count check. Mainly because it's a macro so embedding `#if` it in will break on MSVC (which is actually reasonable  - it follows the C standard here)